native-windows-gui/src/layouts/dyn_layout.rs

```diff
@@ -0,0 +1,344 @@
+        self.update_layout(w as u32, h as u32);
+    }
+
+    /**
+        Check if a window control is a children of the layout
+
+        Panic:
+        - If the layout is not initialized
+        - If the child is not a window-like control
+    */
+    pub fn has_child<W: Into<ControlHandle>>(&self, c: W) -> bool {
+        let inner = self.inner.borrow();
+        if inner.base.is_null() {
+            panic!("DynLayout is not initialized");
+        }
+
+        let handle = c.into().hwnd().expect("Children is not a window-like control (HWND handle)");
+        inner.children.iter().any(|c| c.control == handle )
+    }
+
+    /// Resize the layout as if the parent window had the specified size.
+    ///
+    /// Arguments:
+    ///   w: New width of the layout
+    ///   h: New height of the layout
+    ///
+    ///  Panic:
+    ///   - The layout must have been successfully built otherwise this function will panic.
+    pub fn resize(&self, w: u32, h: u32) {
+        let inner = self.inner.borrow();
+        if inner.base.is_null() {
+            panic!("Layout is not bound to a parent control.")
+        }
+        self.update_layout(w, h);
+    }
+
+    /// Resize the layout to fit the parent window size
+    ///
+    /// Panic:
+    ///   - The layout must have been successfully built otherwise this function will panic.
+    pub fn fit(&self) {
+        let inner = self.inner.borrow();
+        if inner.base.is_null() {
+            panic!("Layout is not bound to a parent control.")
+        }
+
+        let (w, h) = unsafe { wh::get_window_size(inner.base) };
+        self.update_layout(w, h);
+    }
+
+    fn update_layout(&self, width: u32, height: u32) -> () {
+        use winapi::um::winuser::{BeginDeferWindowPos, DeferWindowPos, EndDeferWindowPos};
+        use winapi::um::winuser::{HWND_TOP, SWP_NOZORDER, SWP_NOREPOSITION, SWP_NOACTIVATE, SWP_NOCOPYBITS};
+        use winapi::ctypes::c_int;
+
+        let inner = self.inner.borrow();
+        if inner.base.is_null() || inner.children.len() == 0 {
+            return;
+        }
+
+        let xdelta = 0.01 * width as f32;
+        let ydelta = 0.01 * height as f32;
+
+        unsafe {
+            let hdwp = BeginDeferWindowPos(inner.children.len() as c_int);
+
+            for item in inner.children.iter() {
+                let mut x = item.pos_init.0;
+                if item.mv.0 > 0 { x += (xdelta * item.mv.0 as f32) as i32; }
+
+                let mut y = item.pos_init.1;
+                if item.mv.1 > 0 { y += (ydelta * item.mv.1 as f32) as i32; }
+
+                let mut w = item.size_init.0;
+                if item.sz.0 > 0 { w += (xdelta * item.sz.0 as f32) as i32; }
+
+                let mut h = item.size_init.1;
+                if item.sz.1 > 0 { h += (ydelta * item.sz.1 as f32) as i32; }
+
+                DeferWindowPos(hdwp, item.control, HWND_TOP, x, y, w, h, SWP_NOZORDER | SWP_NOREPOSITION | SWP_NOACTIVATE | SWP_NOCOPYBITS);
```

Comment:
Call `wh::set_window_after(item.control, last_handle)` where last_handle is the handle (`item.control`) of previous control in the iteration. It can also be `None` for the first item.

See https://github.com/gabdube/native-windows-gui/blob/master/native-windows-gui/src/layouts/grid_layout.rs#L463

Without this call, resizing may break the tab order of the controls.